src/toolchain_override.rs

```diff
@@ -33,9 +36,96 @@ pub struct OverrideCfg {
 }
 
 // Represents the [toolchain] table in 'fuel-toolchain.toml'.
-#[derive(Debug, Deserialize, Serialize)]
+#[derive(Debug, Deserialize)]
 pub struct ToolchainCfg {
-    pub channel: String,
+    #[serde(deserialize_with = "deserialize_channel")]
```

Comment:
An alternative to providing a function like this is to implement `Deserialize` for `Channel` manually. This can be useful if `Channel` is used in some other serializable type and we want it to have the same serialization behaviour there too. Not important in this case, just a tip!

Reply:
I saw that there seemed to be more boilerplate code when implementing `Deserialize`, so I opted to do this instead. I'll create an issue for this, it's probably useful for the future as well